Userland/Utilities/disasm.cpp

```diff
@@ -27,7 +28,14 @@ ErrorOr<int> serenity_main(Main::Arguments args)
     args_parser.add_positional_argument(path, "Path to i386 binary file", "path");
     args_parser.parse(args);
 
-    auto file = TRY(Core::MappedFile::map(path));
+    RefPtr<Core::MappedFile> file;
+    u8 const* asm_data = nullptr;
+    size_t asm_size = 0;
+    if ((TRY(Core::System::stat(path))).st_size > 0) {
```

Comment:
Shouldn't we give the user a nice error message if they attempt to disassemble an empty file? With this patch it seems like we "succeed" by doing nothing and pretend everything was alright?

Reply:
I'm not sure. I'm tempted to argue that everything is working as intended, all instructions are disassembled. But on the other hand, you probably didn't mean to disassemble an empty file in >99% of cases, and an error message would make it easier to find out what went wrong. Just a warning on stderr maybe, but still exit with 0?

Reply:
Hmm, no output seems pretty reasonable to me. I checked both of the disassemblers I had installed on my system (`ndisasm` and `objdump`) say nothing when asked to disassemble an empty file. :shrug: